### DISCUSSION TITLE
Hello! Small request

### DISCUSSION THREAD

**Discussion by GianPel90** [Truinto/DarkCodex#23] [Ideas]
Hi! I'm not sure this is the right plase to make a request, I just created an account on GitHub.

First, thank you for this mod!
This implementation of rules and homebrew is fantastic, and even better there is an option to enable/disable them!
Truly, thank you for the time and work you are putting in and the sharing for us lesser capable in modding.

After that I would like to humbly ask a request... should be 'easy' for you after what I've seen but when I tried to do it I just wasted hours.

Since I'm going for a Vital Strike build I was hoping to port my PnP character in the game, but it is missing my historical weapon: the butchering axe:
https://www.d20pfsrd.com/equipment/weapons/weapon-descriptions/axe-butchering/

could you please implement it in the game? O_O

**Comment by Truinto** [comment]
Hello Gian. Lucky I enabled e-mail notification, otherwise I would never notice your message. Usually folks use the 'issues' tab to submit an enhancement ticket. But in the end it doesn't really matter how it's communicated. I do prefer Github, since its well organized. Though a message over Discord would have worked, too.

Your suggestion has a couple of difficulties. I haven't tried to add an exotic weapon yet. Remains to be seen, if there are any issues with picking it from the proficiency feat. Then there is a very unique strength requirement and penalty. Haven't seen that on any weapon. So will need to see if that's doable at all. Lastly, just because a weapon exists, doesn't mean it shows up ingame. Equipment is a bit annoying, since you need to add it to merchants to be actually usable. But that can be made. Just need figure out which lists to use (wrong list and you can only buy it from one guy and only during a specific chapter).

**Reply by GianPel90** [reply]
Wow you actually replied so fast I'm dazzled *_*
Let me say that I'm grateful for whatever effort you are able to put ^_^

If you find too much trouble even a shortcut would be nice... like modifying an existing two handed exotic axe.
This way the weapons should already be present in the world.
Of course a 'clean' solution would be ideal, but I just don't want to distruct you too much from doing what you like with your programming hobby (that I also enjoy!) for my personal benefit

**Comment by Truinto** [comment]
I will try to find time to do this tomorrow. Or at least start with it. Since every weapon is hardcoded, are there any enchantment combination you prefer?
I think I make a vanilla weapon, +1 through +5, and probably a +3 holy butchering axe.

**Reply by GianPel90** [reply]
Generally if one takes the Butchering Axe it's for the Vital Strike build, so the first enchantment should be Impact.
So something like
Vanilla
+1
+1 impact
+2 impact
+3 impact
...
+2 impact holy
 etc

Again thank you for your kindness ^_^

**Comment by Truinto** [comment]
Well, that was sneaky. Impact isn't an enchantment that exists. In any case, I have the code for both Butchering and Impact.
I will need to generate items and bug fix it.

**Reply by GianPel90** [reply]
Damn Owlcat let me down. Didn't think a basic enchantment like that was missing :/
Let me know if you need help with testing, I can start a new game and/or continue the game (I'm in chapter 1).

And again thank you ^_^

**Reply by Truinto** [reply]
[DarkCodex.zip](https://github.com/Truinto/DarkCodex/files/7898824/DarkCodex.zip)

Mind trying out this version. You probably want to use Toybox to spawn in the items. Searching for "Butchering" will work. I did add it to the vendor list RE_Chapter3VendorTableExotic, but I haven't actually found them. I might be to far into the game.

If I do find a better vendor and change the code, I am not entirely sure how that affects saves. Might make it incompatible.

**Reply by GianPel90** [reply]
Great! I'm actually away from home but tomorrow evening should be able to test it!
Thanks, I'll let you know ASAP :)

**Comment by GianPel90** [comment]
Ok I tried a few basic things and everything seems to work correctly... on paper.

1. The Axe is correctly assigned to the Axe Weapon group
2. The -2 penality for insufficient strenght is applied, and vanishes when it is 19 or more
3. The weapon dice damage seems to scale correctly: 
 - A small B.Axe does 2d6
 - A small B.Axe with impact does 3d6
 - A medium B.Axe does 3d6
 - A medium B.Axe with impact does 4d6
 - A large B.Axe does 4d6
 - A large B.Axe with impact does 6d6
 - A huge B.Axe does 6d6
 - A huge B.Axe with impact does 8d6
 - A gargantuan B.Axe does 8d6
 - A gargantuan B.Axe with impact does 12d6
 - A Colossal B.Axe does 12d6
 - A Colossal B.Axe with impact still does 12d6 damage, maybe it's capped but I don't think it will have pratical repercussions.

This is all according to the character sheet. When I go to the initial dungeon underground the city with a level 13 pg to test interaction with Vital Strike and Improved VS... the charater always misses. Well, not really misses, in the combat log it says "Hit" with the correct math but no damage is done, the damage log is absent.

I will try to put a couple of screenshot for clarification.

**Reply by GianPel90** [reply]
![image](https://user-images.githubusercontent.com/97787317/150542881-d9227e13-f7e2-4e8b-b8e7-782019be4ee1.png)

**Reply by Truinto** [reply]
That must be the weirdest bug I have ever created. You hit, but don't deal damage... how?! I will assume that other weapons work just fine.

**Reply by Truinto** [reply]
Thinking about it, maybe there is a null pointer exception during damage calculation. Your game log will be very helpful in that regard.

"%appdata%\..\LocalLow\Owlcat Games\Pathfinder Wrath Of The Righteous"
GameLogFull.txt and Player.log

**Reply by GianPel90** [reply]
[GameLogFull.txt](https://github.com/Truinto/DarkCodex/files/7915877/GameLogFull.txt)
[Player.log](https://github.com/Truinto/DarkCodex/files/7915878/Player.log)

**Reply by Truinto** [reply]
[lib.zip](https://github.com/Truinto/DarkCodex/files/7916834/lib.zip)
Wow, a bit big. Should have said to clear the log before the bug happens and copy it right after. In any case I found the culprit amidst the 17000 exceptions in your log. I think that will fix it. Just replace the dll and try again please.

**Reply by GianPel90** [reply]
Mmm well now the game crashes when I try to open the Inventory.
Also, the character is naked xD

**Comment by GianPel90** [comment]
[GameLogFull.txt](https://github.com/Truinto/DarkCodex/files/7917195/GameLogFull.txt)
[Player.log](https://github.com/Truinto/DarkCodex/files/7917196/Player.log)

Also, Character sheet opens, but not inventory.

When I start a new game there is no problem, but the moment I add an axe the game crashes

**Reply by Truinto** [reply]
Uh... I don't get how m_VisualParameters is so weird. But I figured out how to make it work. I also tried it out. See new release.

I was quite lazy with picking models. So all of the butchering axes look the same. If you find GUIDs of axes that you prefer over this one, let me know. Impact also has no visual effect. I could copy any existing one, if you can think of a good one.

**Reply by GianPel90** [reply]
Yeah this works like a charm, or so it seems!

I will try to tell you if something is funny (I'm still very early in the game and taking my time) but I'll report anything related to your mod that I catch.

Also, again thank you for your kindness in listening to my request!! ^__^
